src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/DefaultCaseDataAccessControl.java

```diff
@@ -420,9 +432,25 @@ private static boolean isRoleAssignmentsOfInterestToAccessProcessCheck(RoleAssig
             || roleAssignment.getGrantType().equals(GrantType.CHALLENGED.name());
     }
 
-    private List<GrantType> generatePostFilteringAccessGrants(List<RoleAssignment> roleAssignments) {
+    private static boolean isRoleAssignmentsOfInterestToAccessGrantCheck(RoleAssignment roleAssignment) {
+        return isRoleAssignmentsOfInterestToAccessProcessCheck(roleAssignment)
+            || roleAssignment.getGrantType().equals(GrantType.BASIC.name());
+    }
+
+    private List<GrantType> generatePostFilteringAccessGrants(CaseDetails caseDetails,
+                                                              List<RoleAssignment> roleAssignments) {
+        String caseState = caseDetails.getState();
+        CaseTypeDefinition caseType = caseDefinitionRepository.getCaseType(caseDetails.getCaseTypeId());
+
         return roleAssignments.stream()
+            .filter(DefaultCaseDataAccessControl::isRoleAssignmentsOfInterestToAccessGrantCheck)
```

Comment:
```suggestion
            .filter(this::isRoleAssignmentsOfInterestToAccessGrantCheck)
```
If we swap to `this` then we don't need to make the methods `static`. (I think I missed this in the original PR when we introduced `isRoleAssignmentsOfInterestToAccessGrantCheck()` that it was `static`)